src/server/JasmineGraphInstanceService.cpp

```diff
@@ -584,6 +587,271 @@ void *instanceservicesession(void *dummyPt) {
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string partitionID = (data);
+            partitionID = utils.trim_copy(partitionID, " \f\n\r\t\v");
+            instance_logger.log("Received Partition ID: " + partitionID, "info");
+
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string workerList = (data);
+            workerList = utils.trim_copy(workerList, " \f\n\r\t\v");
+            instance_logger.log("Received Worker List " + workerList, "info");
+
+            std::vector<string> workerSockets;
+            boost::split(workerSockets, workerList, boost::is_any_of(","));
+
+            // Calculate the out degree distribution in the current super worker.
+            JasmineGraphHashMapLocalStore graphDB;
+            JasmineGraphHashMapCentralStore centralDB;
+            std::map<std::string, JasmineGraphHashMapLocalStore>::iterator it;
+            std::map<std::string, JasmineGraphHashMapCentralStore>::iterator itcen;
+
+            if (JasmineGraphInstanceService::isGraphDBExists(graphID, partitionID)) {
+                JasmineGraphInstanceService::loadLocalStore(graphID, partitionID, graphDBMapLocalStores);
+            }
+
+            if (JasmineGraphInstanceService::isInstanceCentralStoreExists(graphID, partitionID)) {
+                JasmineGraphInstanceService::loadInstanceCentralStore(graphID, partitionID, graphDBMapCentralStores);
+            }
+
+            graphDB = graphDBMapLocalStores[graphID + "_" + partitionID];
+            centralDB = graphDBMapCentralStores[graphID + "_centralstore_" + partitionID];
+
+            map<long,long> degreeDistribution = graphDB.getInDegreeDistributionHashMap();
+            std::map<long,long>::iterator its;
+
+            map<long,long> degreeDistributionCentral = centralDB.getInDegreeDistributionHashMap();
+            std::map<long,long>::iterator itcentral;
+
+            for (its = degreeDistributionCentral.begin(); its != degreeDistributionCentral.end();++its) {
+
+                bool centralNodeFound = false;
+                for (itcentral = degreeDistribution.begin(); itcentral != degreeDistribution.end();++itcentral) {
+
+                    if ((its->first) == (itcentral->first)) {
+                        degreeDistribution[its->first] = (its->second) + (itcentral->second);
+                        centralNodeFound = true;
+                    }
+                }
+                if (!centralNodeFound) {
+                    degreeDistribution.insert(std::make_pair(its->first, its->second));
+                }
+            }
+
+            // Invoke other workers to calculate their own our degree distributions
+            //todo  invoke other workers asynchronously
```

Comment:
When you put a todo comment please follow the convention `//TODO(<your github ID>`:`

Reply:
Fixed

Reply:
Done